classes/robot/crawler.php

```diff
@@ -81,7 +94,8 @@ public function is_bot_valid() {
                 array('resredirect' => $result->redirect));
         }
 
-        $hello = strpos($result->contents, get_string('hellorobot', 'local_linkchecker_robot') . " '{$this->config->botusername}'");
+        $hello = strpos($result->contents,
+            get_string('hellorobot', 'local_linkchecker_robot') . " '" . self::$config->botusername . "'");
```

Comment:
A little off topic but can you pass the bot name into the string as a param and format this inside the lang pack please